### PR TITLE
[bazel] Migrate Google dependencies to bzlmod

### PR DESCRIPTION
**Pull request by jwnrt**
Simple migrations so they can go directly in the main `MODULE.bazel`.

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -4,16 +4,26 @@
 
 module(name = "lowrisc_opentitan")
 
+bazel_dep(name = "abseil-cpp", version = "20230802.0")
 bazel_dep(name = "bazel_skylib", version = "1.5.0")
+bazel_dep(name = "googletest", version = "1.11.0")
+bazel_dep(name = "lowrisc_misc_linters")
```
Comment (nbdd0121) [inline MODULE.bazel:10]: Move this line closer to the override?

Comment (jwnrt) [reply MODULE.bazel:10]: Done

**Comment by jwnrt** [comment]
Sorry for the notifications, this should have been a draft. It's blocked on the buildifier migration.

File: MODULE.bazel (hunk 1 above)
Comment (jwnrt) [reply MODULE.bazel:12]: Would you mind if I keep the dependencies sorted? If I try move this next to the override, `buildifier` tries to move it back into the list but now unsorted.

Comment (nbdd0121) [reply MODULE.bazel:12]: Would buildifier try to move it back to the list even across comments?

Comment (jwnrt) [reply MODULE.bazel]: I'd forgotten to use the `dev_dependency` parameter for this module so I've moved it to a separate section anyway. Now it's next to its override. Let me know if this is clearer nw.

**Review by nbdd0121 — APPROVED** [review]

**Pull request merged**